include/aws_jobs.h

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**@file aws_jobs.h
+ *  The AWS jobs library provides APIs to:
+ *  - Report status
+ *  - Subscribe to job topics
+ *  - String templates for topics
+ *  - Defines for lengths of topics, status and job IDs
+ *  - Defines for Subscribe message IDs
+ *
+ */
+
+#ifndef AWS_JOBS_H__
+#define AWS_JOBS_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <net/mqtt.h>
+
+/* @brief The max JOB_ID_LEN according to AWS docs
+ * https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html
+ */
+#define JOB_ID_MAX_LEN (64)
+#define STATUS_MAX_LEN (12)
+
+/* @brief MQTT message IDs used for identifying subscribe message ACKs */
+#define SUBSCRIBE_ID_BASE (2110)
+#define SUBSCRIBE_NOTIFY (SUBSCRIBE_ID_BASE + 1)
+#define SUBSCRIBE_NOTIFY_NEXT (SUBSCRIBE_ID_BASE + 2)
+#define SUBSCRIBE_GET (SUBSCRIBE_ID_BASE + 3)
+#define SUBSCRIBE_GET_JOB_ID (SUBSCRIBE_ID_BASE + 4)
+#define SUBSCRIBE_JOB_ID_UPDATE (SUBSCRIBE_ID_BASE + 5)
+#define SUBSCRIBE_EXPECTED (SUBSCRIBE_ID_BASE + 10)
+
+/**
+ * @brief String templates for constructing AWS Jobs topics and their max
+ * length
+ */
+#define AWS "$aws/things/"
+#define AWS_LEN (sizeof(AWS) - 1)
+
+#define NOTIFY_TOPIC_TEMPLATE AWS "%s/jobs/notify"
+#define NOTIFY_TOPIC_MAX_LEN (AWS_LEN + CONFIG_CLIENT_ID_LEN + 12)
+
+#define NOTIFY_NEXT_TOPIC_TEMPLATE AWS "%s/jobs/notify-next"
+#define NOTIFY_NEXT_TOPIC_MAX_LEN (AWS_LEN + CONFIG_CLIENT_ID_LEN + 18)
+
+#define GET_TOPIC_TEMPLATE AWS "%s/jobs/get/%s"
+#define GET_TOPIC_LEN (AWS_LEN + CONFIG_CLIENT_ID_LEN + 11)
+
+#define JOB_ID_GET_TOPIC_TEMPLATE AWS "%s/jobs/%s/get/%s"
+#define JOB_ID_GET_TOPIC_MAX_LEN (AWS_LEN +\
+				 CONFIG_CLIENT_ID_LEN +\
+				 JOB_ID_MAX_LEN +\
+				 12)
+
+#define JOB_ID_UPDATE_TOPIC_TEMPLATE AWS "%s/jobs/%s/update"
+#define JOB_ID_UPDATE_TOPIC_MAX_LEN (AWS_LEN +\
+				     CONFIG_CLIENT_ID_LEN +\
+				     JOB_ID_MAX_LEN +\
+				     14)
+
+
+/** @brief Job Execution Status. */
+enum execution_status {
+	QUEUED = 0,
+	IN_PROGRESS,
+	SUCCEEDED,
+	FAILED,
+	TIMED_OUT,
+	REJECTED,
+	REMOVED,
+	CANCELED
+};
+
+/** @brief Mapping of enum to strings for Job Execution Status. */
+const char *execution_status_map[] = {
+	"QUEUED",
+	"IN_PROGRESS",
+	"SUCCEEDED",
+	"FAILED",
+	"TIMED_OUT",
+	"REJECTED",
+	"REMOVED",
+	"CANCELED"
+};
+
+/**
+ * @brief Constructs then subscribes to get topic for AWS IoT jobs for accepted
```

Comment:
Needs to be rewritten.

Reply:
for all construct comments?

Reply:
The sentence is incomplete IMO, only this and the one mentioned above.